nilearn/region_extraction/regionextractor.py

```diff
@@ -0,0 +1,341 @@
+        if self.extractor not in extractor_methods:
+            message = ('"extractor" should be given '
+                       'either of these {0}').format(extractor_methods)
+            raise ValueError(message)
+
+        if isinstance(self.threshold, float):
+            ratio = self.threshold
+        elif self.threshold == 'auto':
+            ratio = 0.8
+        elif self.threshold is not None:
+            raise ValueError('Threshold must be given as '
+                             '"auto" or float. You have given %s. '
+                             % str(self.threshold))
+
+        if self.threshold_strategy == 'voxelratio':
+            raveled = np.abs(transformed_maps).ravel()
+            argsort = np.argsort(raveled)
+            n_voxels = (ratio * transformed_maps.size)
+            threshold = raveled[argsort[- n_voxels]]
+            transformed_maps[np.abs(transformed_maps) < threshold] = 0.
+        elif self.threshold_strategy == 'percentile':
+            percentile = 100 - (100 / len_maps) * ratio
+            threshold = scoreatpercentile(
+                np.abs(transformed_maps), percentile)
+            transformed_maps[np.abs(transformed_maps) < threshold] = 0.
+
+        regions_accumulated = []
+        index = []
+
+        for i, trans_map in enumerate(transformed_maps):
+            each_map_data = maps_data[..., i]
+            trans_map_img = self.masker_.inverse_transform(trans_map)
+            trans_map_data = trans_map_img.get_data()
+
+            if self.extractor == 'voxel_wise':
+                label_maps, n_labels = label(trans_map_data)
+                regions = extract_regions(each_map_data, affine,
+                                          label_maps, min_size)
+                regions_accumulated.append(regions)
+            elif self.extractor == 'local_regions':
+                smooth_fwhm = self.smooth_fwhm
+                smooth_data = _smooth_array(trans_map_data,
+                                            affine, fwhm=smooth_fwhm)
+                seeds = peak_local_max(smooth_data, indices=False,
+                                       exclude_border=False)
+                seeds_label, seeds_id = label(seeds)
+                # Assigning "-1" as ignored area to random walker
+                seeds_label[trans_map_data == 0] = -1
+                seeds_map = random_walker(trans_map_data, seeds_label,
+                                          mode='cg_mg')
+                # Again replace "-1" values with "0" for an expected behaviour
+                # to region seperation
+                seeds_map[seeds_map == -1] = 0
+                seeds_label_maps, n_seeds_labels = label(seeds_map)
+                regions = extract_regions(each_map_data, affine,
+                                          seeds_label_maps, min_size)
+                regions_accumulated.append(regions)
+
+        all_regions_imgs = concat_niimgs(regions_accumulated)
```

Comment:
I think that this won't work as you exemple since `regions_accumulated is a list of list of Nifti1Image. You should obtain a 5D nifti.